### PR TITLE
[AKS] add POST endpoint for fetching accessProfiles

### PR DESCRIPTION
**Pull request by mboersma**
Adds a new POST endpoint that is preferred to GET for fetching AKS access profiles (Kubernetes configs). The endpoint is already deployed to RPs in all AKS regions.

I've tested this with the Python SDK and simple CLI changes that are needed and it works well.

cc: @weinong @shrutir25 @rjtsdl @sauryadas

### PR information
- [X] The title of the PR is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [X] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [X] If applicable, the PR references the bug/issue that it fixes.
- [X] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [X] My spec meets the review criteria:
  - [X] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [X] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/containerservices/resource-manager/Microsoft.ContainerService/stable/2017-08-31/managedClusters.json [hunk 1]
```diff
@@ -154,6 +154,58 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerService/managedClusters/{resourceName}/accessProfiles/{roleName}/listCredential": {
+      "post": {
```
[the remark this answers is not in the document]
Comment (mboersma) [reply specification/containerservices/resource-manager/Microsoft.ContainerService/stable/2017-08-31/managedClusters.json]: @lmazuel done.

**Review by hovsepm — APPROVED** [review]

**Review by hovsepm — CHANGES_REQUESTED** [review]
Operation name with prefix List assumes multiple return values in array. You already have "operationId": "ManagedClusters_GetAccessProfiles", which returns exactly the same type `ManagedClusterAccessProfile`. then what is a point to add this new endpoint with such a confusing method name?

**Comment by mboersma** [comment]
> what is a point to add this new endpoint with such a confusing method name?

@hovsepm I was just using the name defined for the endpoint in the AKS RP. I agree it's not good terminology though. We can call it something different in the swagger if that's preferable. The idea here is we'd prefer POST to GET for this sensitive endpoint.

@weinong anything to add?

**Comment by hovsepm** [comment]
@mboersma yes please change the operation name. it is used to generate a method names in the SDKs in each language. Name should be self descriptive and discoverable. Please also revise the description because now for this 2 exposed operations
```
...
"operationId": "ManagedClusters_ListCredential",
"description": "Gets the accessProfile for the specified role name of the managed cluster with a specified resource group and name."
...
"operationId": "ManagedClusters_GetAccessProfiles"
"description": "Gets the accessProfile for the specified role name of the managed cluster with a specified resource group and name."
```
as a user of the generated SDK - how should I understand which method to call? What is the difference between them?

File: specification/containerservices/resource-manager/Microsoft.ContainerService/stable/2017-08-31/managedClusters.json [hunk 2]
```diff
@@ -116,7 +116,8 @@
         ],
         "operationId": "ManagedClusters_GetUpgradeProfile",
         "summary": "Gets upgrade profile for a managed cluster.",
-        "description": "Gets the details of the upgrade profile for a managed cluster with a specified resource group and name.",
+        "deprecated": true,
+        "description": "Use ManagedClusters_ListCredential instead.",
```
Comment (hovsepm) [inline specification/containerservices/resource-manager/Microsoft.ContainerService/stable/2017-08-31/managedClusters.json]: I think you are deprecating the wrong operation.  I believe you should deprecate operationId `ManagedClusters_GetAccessProfiles`, right?

Comment (mboersma) [reply specification/containerservices/resource-manager/Microsoft.ContainerService/stable/2017-08-31/managedClusters.json]: D'oh, my mistake. I'll fix.

**Review by hovsepm — APPROVED** [review]

**Comment by hovsepm** [comment]
@mboersma I'll merge this PR however there is an issue that needs to be fixed for each endpoint. Please take a look to https://github.com/Azure/azure-rest-api-specs/issues/3113

**Comment by tombuildsstuff** [comment]
@mboersma 👋 out of interest, is there a timeframe for deprecating the old (GET) endpoint?

**Comment by mboersma** [comment]
@tombuildsstuff once this change makes it into the `az` CLI, we plan to start returning a (useful) error from the GET endpoint 14 days afterward. So most likely that means two weeks after June 5. Let me know if that's going to cause problems!

**Comment by tombuildsstuff** [comment]
@mboersma given that won't be available in the Go SDK until around that date, I don't feel that's a long enough period for deprecation IMO - we (Terraform) expose these values on [our AKS resources](https://www.terraform.io/docs/providers/azurerm/r/kubernetes_cluster.html), which we consume via the Azure SDK for Go. Whilst we're probably able to time a release shortly after this change goes live - however we should give users some time to update to the new version which uses Post instead of Get.

As such - would it be possible for the old endpoint to continue working until the start of July? Whilst I can understand this isn't ideal from your side, I think it's better to not break users workflows without a couple of weeks extra transition period for downstream dependencies once the SDK's are out :)

Thanks!

**Comment by tombuildsstuff** [comment]
cc @joshgav / @marstr / @jhendrixMSFT :)

**Comment by mboersma** [comment]
>  I don't feel that's a long enough period for deprecation IMO

Honestly, I don't think that's long enough either, I'm just reporting the current intent. I'll bring your feedback to the team and see if we can make the deprecation timeline more humane.

**Comment by tombuildsstuff** [comment]
@mboersma awesome, thanks :)

**Comment by mboersma** [comment]
@tombuildsstuff we will comply with the standard Azure deprecation policy and keep the GET endpoint functional for 12 more months. (We got similar feedback from ARM review on the newer API in #3131.)

**Comment by tombuildsstuff** [comment]
@mboersma that's awesome, thanks so much :)

**Pull request merged**